src/com/hms/flexyosisoftconnector/Tag.java

```diff
@@ -120,6 +170,264 @@ public void recordTagValue(String time)
+
+               // if last data point is not null, check the new one is new
+               if (lastDataPoint == null) {
+                   // there was no last point, start of program condition
+                   recordTagValue(timeInt, value);
+               } else {
+                   // convert last point's epoch time to seconds, which is what the EBD "timeInt" is in
+                   long lastPointSeconds = lastDataPoint.epochTime/1000;
+                   if (lastPointSeconds != Long.parseLong(timeInt)) {
+                       recordTagValue(timeInt, value);
+                   }
+               }
+
+               // read next line, repeat
+               line = reader.readLine();
+           }
+       } catch (IOException e) {
+           Logger.LOG_ERR("IO exception in Tag Class, reading in historical log file.");
+           e.printStackTrace();
+       }
+   }
+
+
+   /**
+    * converts given long into date string of expected format
+    * and writes it to a file
+    *
+    * @param timeInt epoch time in seconds to get converted and stored in file
+    * @return just return what string it set
+    */
+   public String writeTimePointerFile(long timeInt)
+   {
+      // convert from epoch time long to string
+      String fileValueString = convertTimeIntHelper(timeInt);
+
+
+      File file = new File(tagTimeIntFileName);
+      FileWriter fr = null;
+      BufferedWriter br = null;
+      try{
+          fr = new FileWriter(file);
+          br = new BufferedWriter(fr);
+
+          // write the data to the file here
+          br.write(fileValueString);
+
+      } catch (IOException e) {
+          e.printStackTrace();
+      }finally{
+          try {
+              br.close();
+              fr.close();
+          } catch (IOException e) {
+              e.printStackTrace();
+          }
+      }
+      return fileValueString;
+   }
+
+   /**
+    * This function is only called when retrieving value for call to export block descriptor.
+    *
+    * the tagTimeIntFileName file holds a time pointer for every tag. 1 file exists per tag.
+    * this will hold just a string which will be the epoch time in seconds for where/when
+    * in the Flexy historical logs has the data queue left off on sending data to OSIsoft
+    *
+    * The file can only contain numbers because it is converted to a long
+    *
+    * The below function makes sure the file exists, if not add in the current time to that
+    * otherwise read off the epoch time in that file
+    *
+    * @return Returns a string of the file contents. it will be the epoch time pointer for the data queue.
+    * also called a timeInt in the return file from the EBD commands. name will be timeInt throughout code.
+    */
+   public String readTimePointerFile()
+   {
+      String timeInt = "-1";
+      // check if file exists
+      File tempFile = new File(tagTimeIntFileName);
+      boolean exists = tempFile.exists();

```

Comment:
consider a better name for this boolean, it is not descriptive as to what exists.